src/custom/pages/Swap/SwapMod.tsx

```diff
@@ -380,11 +380,47 @@ export default function Swap({
       approveRequired = true
     }
 
-    reportAnalytics('Approve', v2Trade?.inputAmount?.currency.symbol)
     if (approveRequired) {
-      return approveCallback().catch((error) => console.error('Error setting the allowance for token', error))
+      const symbol = v2Trade?.inputAmount?.currency.symbol
+      reportAnalytics('Send Token Approval to Wallet', symbol)
+      return approveCallback()
```

Comment:
Why not write this as:
```ts
let swapErrorMessage = 'Sign Token Approval', actionAnalytics = symbol, errorCode
return approveCallback()
    .catch((error) => {
        console.error('Error setting the allowance for token', error)
        if (isRejectRequestProviderError(error)) {
        swapErrorMessage = 'User rejected approving the token'
        actionAnalytics = 'Reject Token Approval'
        } else {
        swapErrorMessage = getProviderErrorMessage(error)
        actionAnalytics = 'Signing Error for Token Approval'
        
        if (error?.code && typeof error.code === 'number') {
          errorCode = error.code
        }
      }

        setSwapState({
            attemptingTxn: false,
            tradeToConfirm,
            showConfirm,
            swapErrorMessage,
            txHash: undefined,
        })
    })
    .finally(() => reportAnalytics(actionAnalytics, symbol, errorCode))
```

Reply:
It's a good suggestion, but I would prefer to keep it like this, just to not rely on the mutation of some`swapErrorMessage` outside of the promise